client/src/Components/Registration/RegistrationPage.jsx

```diff
@@ -108,13 +108,56 @@ class RegistrationPage extends Component {
   passwordComplexity = e => {
     const password = e.target.value;
     const checks = complexityCheck(password);
-    // furture - using dynamic checks, more elaborate failure messages could be included
-    if (!checks.validPW) {
+    const {
+      length,
+      validChar,
+      capital,
+      lowercase,
+      number,
+      specChar,
+      validPW
+    } = checks;
+    let lengthErr,
+      validCharErr,
+      capitalErr,
+      lowercaseErr,
+      numberErr,
+      specCharErr;
+    // --Error message builder--
+    if (!validPW) {
+      // First define all relevant error messages
+      if (!length) lengthErr = <div> - contain at least 8 chars</div>;
+      if (!validChar)
+        validCharErr = <div> - contain only valid characters</div>;
+      if (!lowercase)
+        lowercaseErr = <div> - contain at least one lowercase letter</div>;
+      if (!capital)
+        capitalErr = <div> - contain at least one capital letter</div>;
+      if (!specChar)
+        specCharErr = <div> - contain one of !,#,$,%,&,?,@,^,~</div>;
```

Comment:
This line and the opening < on line 141 are flagging as red. Perhaps these need to be surrounded by backpacks or something to prevent characters causing unexpected code issues?

Reply:
Done. Good catch. It wasn't causing errors, but I remember thinking when I typed it that it should.